gateway/managers/syncman/store_kube.go

```diff
@@ -386,10 +255,25 @@ func (s *KubeStore) DeleteProject(ctx context.Context, projectID string) error {
 	return err
 }
 
-func (s *KubeStore) getProjects(v map[string]*config.Project) []*config.Project {
-	projects := []*config.Project{}
-	for _, value := range v {
-		projects = append(projects, value)
+// GetGlobalConfig gets config of all resource required by a cluster
+func (s *KubeStore) GetGlobalConfig() (*config.Config, error) {
```

Comment:
I suggest you get just the project config first. Then you should get each config one by one. This will guarantee you get the configmap in the right order. Also, it will reduce the pressure on the List function.

You can make an array of resource type in the right order and loop on it to fetch each config one by one. Make the array in the `config` package. It can be useful in other places too.